Tools/LambdaTestTool-v2/tests/Amazon.Lambda.TestTool.UnitTests/Services/ApiGatewayRouteConfigServiceTests.cs

```diff
@@ -0,0 +1,158 @@
+using System.Text.Json;
+using Amazon.Lambda.TestTool.Models;
+using Amazon.Lambda.TestTool.Services;
+using Amazon.Lambda.TestTool.Services.IO;
+using Microsoft.Extensions.Logging;
+using Moq;
+
+namespace Amazon.Lambda.TestTool.UnitTests.Services;
+
+public class ApiGatewayRouteConfigServiceTests
```

Comment:
You need tests with `{proxy+}` especially when there are other routes. And you need tests with some overlap in the routes. For example a test with routeA -> `/foo/{value}` and routeB -> `/foo/bar` and you test with requests to `/foo`, `/foo/bar` and `/foo/fred`. Similar idea with `{proxy+}` and make sure we mimic API Gateway's behavior when choosing to use the proxy endpoint versus a more direct path. Another example routeA `/{proxy+}` and routeB `/phil`. Requests going to `/norm` and `/phil` and `/phil/dotnet`. Don't expect the Microsoft template library to match API Gateway's behavior.

Reply:
I have updated the implementation to account for it as well as add sorting based on the route priority logic from API Gateway. I have added comments to explain everything in the code.
I also added 1 big test that lists a bunch of different routes with overlap and tests which function was matched. This should cover everything.